### ISSUE TITLE
VALIDATE_ALL_CODEBASE: false appears not to work. Is there something I'm missing?

### ISSUE THREAD

**Issue by mbuotidem** [github/super-linter#728]
My understanding of this flag is that if set, super-linter will run its linters on existing files that have been modified in the commit. However, as you can see in https://github.com/mbuotidem/lint-demo, it doesn't seem like that is the case. I started out the repo with `existing_code.py`, then added the super-linter action. After that, I edited `existing_code.py`. However, super-linter did not lint it. I wondered if that was perhaps because I did not provide config files for the linters I selected and so I did so, providing config files for pylint, flake8 and black. That didn't help. Linting only occurred when I changed `VALIDATE_ALL_CODEBASE: false` to `VALIDATE_ALL_CODEBASE: true`. 

Is there something I'm missing? Thanks.

**Comment by GaboFDC** [comment]
I think you are miss understanding.

From the readme:
> VALIDATE_ALL_CODEBASE: Will parse the entire repository and find all files to validate across all types. NOTE: When set to false, only new or edited files will be parsed for validation.

**Comment by mbuotidem** [comment]
@GaboFDC  Thanks for responding. I did read that section of the readme before asking the question and I thought I understood it. So perhaps you could clarify to help me and others who could possibly arrive at the same misunderstanding. 

My possible misunderstanding of what `VALIDATE_ALL_CODEBASE` does is based on https://github.com/github/super-linter/issues/219#issuecomment-654187666 and https://github.com/github/super-linter/issues/424#issuecomment-663041562 . Both comments as well as the portion of the readme you quoted seem to suggest that with `VALIDATE_ALL_CODEBASE` set to `false`, edited or modified files as well as newly added files will be validated. It doesn't seem like that's happening on my repo hence my confusion. Again, I'm likely misunderstanding how all this works so I appreciate any insight. To help me be sure that I'm thinking of this properly, here are some noob questions. I apologize if the answers seem obvious but sometimes what's obvious to one person isn't so obvious to a newcomer.

If I have an existing repo and then add the super-linter action to it with `VALIDATE_ALL_CODEBASE : false`, super-linter will ignore the pre-existing files and only run on newly added files, right? 

If I were to make an edit to a pre-exisiting file and the flag is still set at `VALIDATE_ALL_CODEBASE: false`, it will run the linting validation on the recently edited file, right? 

And finally, if I were to add an entirely brand new file to the repo with the flag still at `VALIDATE_ALL_CODEBASE: false`, it would run the validation on the newly added file, right? 

If the answer to any of the above questions is true, then it would appear that something is wrong or more likely, I am misconfiguring super linter. If you check out the linked repo https://github.com/mbuotidem/lint-demo, you will see 7 commits. Each commited .py file has obvious lint errors that any of the linters would catch if they ran.

Commit 1 - Added a file called `existing_code.py`

Commit 2 - Added the super-linter action with :
```
VALIDATE_ALL_CODEBASE : false 
VALIDATE_PYTHON_PYLINT: true 
VALIDATE_PYTHON_FLAKE8: true 
VALIDATE_PYTHON_BLACK: true
```
Commit 3 - Modified `exisiting_code.py` (in my understanding, super-linter should attempt to run validation on this edited file but it didn't)

Commit 4 - Since super-linter did not run validation so I added config files in case that was the issue.  (still no validation )

Commit 5 - Changed `VALIDATE_ALL_CODEBASE: false` to `VALIDATE_ALL_CODEBASE : true` (now validation ran)

Commit 6 - Set `VALIDATE_ALL_CODEBASE` back to false (no validation)

Commit 7 - Added a brand new python file `new_code.py` to see if it would validate it (with the flag back at `VALIDATE_ALL_CODE:false` validation still did not run on the newly created file)

**Comment by leoalves100** [comment]
@mbuotidem 

Translated by google translator.

I also had the same problem. The only solution is to leave **VALIDATE_ALL_CODEBASE: true** and the other types that you want to validate with **true** also.

**Code**
![](https://i.ibb.co/TkMgk6G/Captura-de-tela-2020-09-22-014218.png)

**Execution**
![](https://i.ibb.co/jLT10Qd/Captura-de-tela-2020-09-22-014339.png)

Checked only the files I wanted!

**Comment by mbuotidem** [comment]
@leoalves100  Thanks for your response. Good to see I wasn't the only one who was confused by this. Your answer will probably be helpful for someone else who has the same issue. I also noticed that **VALIDATE_ALL_CODEBASE: false** worked fine when I performed pull requests instead of direct pushes to master. That was going to be the standard method of usage for the project I wanted to implement this on anyway. 

I don't know if it is a bug or intended behavior for super-linter that direct pushes to master will not trigger the validation for the linter but since it works with pull requests which is how I'm using this, I'll close this question.

**Comment by hepcat72** [comment]
I read all this (and what I thought were the relevant portions of the doc) and I still didn't understand.  Linting of our code base takes about a half hour every time, even if you change just one file.  I found the `VALIDATE_ALL_CODEBASE` setting and thought it would be able to save me loads of time.  The default is true, so I figured (since I wasn't setting it) setting it to false would result in only the edited/added files to be linted, yet it still linted everything.  I have learned (and this is documented, although not cross-referenced from the `VALIDATE_ALL_CODEBASE` description) that setting `RUN_LOCAL` sets `VALIDATE_ALL_CODEBASE` to true regardless of how you explicitly set it).

So for anyone else arriving here and being confused by the above when running superlinter locally, note that to only check changed files, you can set `FILTER_REGEX_INCLUDE` explicitly with the changed/added files, so I wrote a shell script that grabs the changed/added files and sets that variable in the local run.
